fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -1756,6 +1762,80 @@ private Bundle processEntriesForMethod(Bundle requestBundle, Bundle responseBund
         }
     }
 
+    /**
+     * Processes a request entry with a request method of Patch.
+     *
+     * @param requestEntry
+     *            the request bundle entry
+     * @param responseEntry
+     *            the response bundle entry
+     * @param responseIndexAndEntries
+     *            the hashmap containing bundle entry indexes and their associated response entries
+     * @param entryIndex
+     *            the bundle entry index of the bundle entry being processed
+     * @param localRefMap
+     *            the map of local references to external references
+     * @param requestURL
+     *            the request URL
+     * @param absoluteUri
+     *            the absolute URI
+     * @param requestDescription
+     *            a description of the request
+     * @param initialTime
+     *            the time the bundle entry processing started
+     * @throws Exception
+     */
+    private void processEntryforPatch(Entry requestEntry, Entry responseEntry, Map<Integer, Entry> responseIndexAndEntries, Integer entryIndex,
+        Map<String, String> localRefMap, FHIRUrlParser requestURL, String absoluteUri, String requestDescription, long initialTime)
+        throws FHIROperationException {
+        FHIRRestOperationResponse ior = null;
+        int resourceIdIndex = 1;
+        int resourceTypeIndex = 0;
+        String resourceType = extractValueFromUrl(requestEntry.getRequest().getUrl().getValue(), resourceTypeIndex);
+        String resourceId = extractValueFromUrl(requestEntry.getRequest().getUrl().getValue(), resourceIdIndex);
+        try {
+            if (resourceType != null && resourceId != null) {
+
+                if (requestEntry.getResource().is(Parameters.class)) {
```

Comment:
There are a couple of 'if' checks here. Do we want to issue an error in the case where resourceType or resourceId is null, or if the requestEntry.getResource() is not of type Parameters?

Reply:
could you please suggest how can we handle? in case the resourceType && resourceId are null,
What if the requestEntry.resource is not of parameter class Type

Reply:
I would handle similar to what's done in processEntryForPut. In this case, since we're not implementing the conditional update, only a requestUrl with both resourceType and resourceId is valid:

```
        String[] pathTokens = requestURL.getPathTokens();
        String resourceType = null;
        String resourceId = null;

        // Process a PATCH.
        if (pathTokens.length == 1) {
            // A single-part url would be a conditional update: <type>?<query>
            // This is not yet supported for PATCH requests.
            String msg = "Conditional update operation is not supported for PATCH requests.";
            throw buildRestException(msg, IssueType.NOT_SUPPORTED);
        } else if (pathTokens.length == 2) {
            // A two-part url would be a normal patch: <type>/<id>.
            resourceType = pathTokens[0];
            resourceId = pathTokens[1];
        } else {
            // A url with any other pattern is an error.
            String msg = "Request URL for bundled PATCH request should have path part with two tokens (<resourceType>/<id>).";
            throw buildRestException(msg, IssueType.INVALID);
        }
```

In your try block, you then just have to check if resource type is Parameters. If it's not, I would suggest throwing an exception with IssueType.INVALID and a message of "Request resource type for PATCH request must be type 'Parameters'."

Reply:
Thank you for the suggestion